### ISSUE TITLE
research: benchmark environment-level RF world embeddings against MERIDIAN site adaptation

### ISSUE THREAD

**Issue by ruvnet** [ruvnet/RuView#1701]
## Problem

ADR-027 MERIDIAN treats the environment primarily as a nuisance domain to factor out of pose features. New work suggests the environment should also be modeled explicitly as a reusable latent object.

Channel2World (arXiv:2608.17544, submitted 2026-08-18) learns a fixed environment embedding from multiple MIMO channel and position observations. The frozen environment encoder then conditions localization, sparse beam-domain CSI reconstruction, and RF-observable geometry reconstruction in unseen environments without gradient-based site fine-tuning.

This does not invalidate MERIDIAN. It changes the decomposition:

* human/task features should remain environment-invariant where possible
* propagation/environment state should become an explicit reusable world prior rather than discarded training residue

The paper is simulation-only and no official code was verified, so this should begin as a bounded reproduction and benchmark, not a production dependency.

## Current architecture

ADR-027 uses a `HardwareNormalizer`, domain factorizer, environment encoder used mainly as a training signal, and geometry-conditioned pose inference. It currently proposes resampling heterogeneous hardware into a canonical 56-subcarrier representation.

RuView also now has:

* multistatic CSI time alignment
* active sensing and information-gain packages
* spatial/world-model packages
* RuVector-backed memory paths
* iPhone LiDAR geometry capture for future ground truth and fusion

## Observed limitations

1. Environment information is treated mainly as shortcut information to suppress rather than a reusable physical prior.
2. Canonical subcarrier resampling can destroy bandwidth, phase, antenna, and hardware information needed for multipath/path-token learning.
3. Site adaptation is still framed around task labels or environment-specific adaptation rather than conditioning from unlabeled channel observations.
4. There is no explicit benchmark comparing environment embedding against fine-tuning at equal context budgets.

## Research evidence

Primary: Channel2World, https://arxiv.org/abs/2608.17544

Reported setup:

* 26,000 ray-traced base-station-centered environments
* approximately 5,000 channel measurements per environment
* 128 context channels and 32 query channels in the reported configuration
* 16 latent environment tokens, embedding dimension 128
* approximately 9.2M model parameters
* downstream evaluation on localization, sparse beam CSI reconstruction, and RF-observable geometry

Important boundary: the paper validates on ray-tracing data. It explicitly leaves measured multi-site validation as future work. No production claim should be derived from the reported results.

Related deployment review: Wireless Physical-Layer Foundation Models, arXiv:2608.20486, emphasizes held-out sites, bands, arrays, hardware, mobility, latency, energy, fallback, and hybrid classical/learned operation.

## Proposed architecture

Add an experimental environment-conditioning path beside MERIDIAN rather than replacing it.

```text
native complex RF observations + sensor poses
            |
            v
   multipath/path extractor
      AoA / delay / gain
            |
            v
   EnvironmentEncoder
            |
            v
   RfEnvironmentEmbedding
      frozen reusable prior
            |
      +-----+------------------+
      |                        |
 localization head       CSI reconstruction head
      |                        |
 pose/task head           RF geometry head
```

Store the environment embedding as residual latent state in RuVector. Authoritative geometry, calibration, provenance, and device state remain symbolic and receipt-backed. The latent must never replace known facts.

## Target packages

Prefer existing packages. Confirm exact ownership before code changes.

Likely areas:

* MERIDIAN / domain-generalization training path
* existing CSI encoder package
* `ruview-memory` or current RuVector integration for environment embeddings
* `ruview-groundtruth` for measured geometry labels
* `ruview-bench` or current benchmark harness

Do not create a new repository.

## Data contract

Prototype:

```rust
pub struct RfEnvironmentEmbedding {
    pub environment_id: String,
    pub vector: Vec<f32>,
    pub context_event_ids: Vec<String>,
    pub context_count: u32,
    pub model_id: String,
    pub calibration_ids: Vec<String>,
    pub created_ns: u64,
    pub uncertainty: f32,
}
```

The authoritative input remains native complex RF plus explicit geometry. Do not make canonical 56-subcarrier data the only retained representation.

## Benchmark design

Compare four adaptation strategies under identical context budgets:

A. current frozen baseline

B. MERIDIAN geometry conditioning

C. site-specific fine-tuning

D. frozen environment embedding conditioning

Context budgets: 16, 32, 64, 128, 256, and 1024 observations where available.

Hold out complete rooms, buildings, hardware families, antenna layouts, days, and people where relevant.

Primary metrics:

* localization p50/p90 error
* CSI reconstruction NMSE
* environment geometry error where ground truth exists
* adaptation wall-clock time
* GPU memory
* model size
* context bytes
* inference p50/p95 latency
* calibration error / selective risk

## Real-world validation

Synthetic Sionna/ray-traced data may be used for pretraining only.

The promotion gate must use captured or live real data from at least:

* 5 unseen rooms
* 2 buildings
* 3 RF hardware families if available
* 2 antenna layouts
* multiple capture days

The new iPhone LiDAR path can provide metric geometry for a subset of rooms. It must be used as ground truth or calibration evidence, not silently as a runtime feature when evaluating RF-only performance.

## MetaHarness configuration

Use the repository-native optimization manifest. Only mutate objectively benchmarkable parameters such as context count, token count, latent width, masking ratio, and adapter size.

Hard vetoes:

* any cross-environment regression beyond the agreed bound
* non-disjoint train/test environment IDs
* higher privacy exposure
* loss of provenance coverage
* latency or memory outside the deployment budget
* synthetic-only gain presented as field gain

Darwin optimization is allowed only after the deterministic benchmark and data-split manifest are fixed.

## Security and privacy

Environment embeddings can encode occupancy patterns, room identity, device fingerprints, and potentially sensitive location information.

Controls:

* derive embeddings at the edge by default
* scope them to tenant and physical site
* never allow a raw RF frame to cross a P0 boundary merely because an embedding is lower class
* bind embeddings to source-event receipts
* model poisoning tests using adversarial/malformed RF windows
* bounds on context count and tensor dimensions
* reject NaN/Inf
* model artifact hashes and signed weights
* no external text from research datasets enters agent prompts as trusted instructions

## Dependencies and licensing

Do not copy Channel2World code unless an official repository and compatible license are verified. Initial reproduction should be clean-room from the paper using existing RuView/RuVector components.

## Expected measurable benefit

Success is not “better embeddings.” It is one or more of:

* equal or better unseen-room localization with at least 4x fewer labeled adaptation examples
* at least 50% reduction in site adaptation wall-clock time
* better sparse CSI reconstruction at a fixed pilot/context budget
* useful RF-observable geometry prediction on held-out real rooms

No benefit is claimed until measured.

## Backward compatibility

Experimental feature flag only. Existing MERIDIAN path remains default. No wire-format change is required for the initial benchmark.

## Rollback

Disable the environment-conditioning feature and delete its stored latent artifacts. Existing symbolic geometry, RF frames, and MERIDIAN models remain valid.

## Definition of done

A reproducible report compares A/B/C/D on fixed disjoint splits with at least five runs for stochastic training, includes hardware/OS/compiler/runtime/model/dataset/configuration, and demonstrates whether environment conditioning improves a real held-out site metric without violating latency, memory, privacy, or provenance gates.

**Comment by ruvnet** [comment]
## 2026-09-07 SOTA delta: couple RF world state to communication scheduling

New evidence strengthens this issue without justifying a new subsystem.

`Coupled Control and Wireless World Models for Resilient Remote Robotic Control` (arXiv:2609.04851v1, submitted 2026-09-04, CC0) couples a control JEPA with a wireless JEPA over raw CSI plus structured RF representations. In synchronized Gazebo, ROS, and Sionna simulation, the originating team reports:

* uplink latency 0.48 ± 0.02 ms versus 0.86 ms DQN and 2.63 ms PID
* total uplink energy 4.74 ± 0.16 J versus 9.21 ± 0.32 J DQN and 11.19 ± 0.15 J PID
* 100% navigation completion after access-point handover versus 56.48% DQN and 58.99% PID
* spectrogram and persistence-image RF representations reduce W-JEPA prediction error versus raw CSI across the tested prediction horizons
* spectrogram generation around 1.6 to 1.7 ms in the reported configurations, while persistence-image generation is roughly 99 to 104 ms, exposing a material edge-runtime tradeoff

Evidence class: originating-team measured in simulation. The paper explicitly limits the current result to one robot, one wireless link, predefined navigation scenarios, and simulation, with physical validation left as future work.

### Update the existing benchmark rather than create a parallel world model

Add two optional heads to condition D in this issue:

1. wireless-state forecast: predict future channel quality or transmission-power requirement from the existing environment latent
2. communication scheduler: decide whether the next high-dimensional sensor update is necessary under a frozen control-quality bound

Add a structured-RF ablation:

A. native complex CSI only

B. CSI plus spectrogram

C. CSI plus persistence-image representation

D. CSI plus the cheapest structured representation that survives the edge budget

### Additional metrics

Record sensor uplink bytes, transmissions per control step, p50/p95 communication latency, radio energy, representation-generation latency, handover completion, packet-loss resilience, and downstream navigation or tracking quality.

### Additional promotion gate

No communication-aware path advances until real hardware reproduces at least one material systems gain while preserving the task metric. Initial target: at least 30% fewer transmitted sensor bytes or 30% lower radio energy, task success within 2 absolute points of the stronger baseline, p95 added representation latency below the deployment budget, and no worse recovery after AP handover or packet-loss bursts.

Do not import the external implementation yet. Reuse current RuView world-model, RuVector memory, RuField, and benchmark primitives. Existing MERIDIAN and raw-CSI paths remain rollback.
